src/platform/assets/components/AssetBrowserModal.vue

```diff
@@ -88,8 +98,10 @@ function handleClose() {
   emit('close')
 }
 
-async function handleAssetSelectAndEmit(asset: AssetDisplayItem) {
+function handleAssetSelectAndEmit(asset: AssetDisplayItem) {
   emit('asset-select', asset)
-  await selectAssetWithCallback(asset.id, props.onSelect)
+  // onSelect callback is provided by dialog composable layer
+  // It handles the appropriate transformation (filename extraction or full asset)
+  props.onSelect?.(asset)
```

Comment:
No need to change it here, but I don't like the two ways of reacting from above:
- Listening for `@asset-select`
- Passing down a callback